bookdown/01-mlr3-basics.Rmd

```diff
@@ -699,17 +699,94 @@ lrn
 
 ## Binary classification {#binary}
 
-### ROC
+Classification problems with only two classes within the target variable are called **binary classification**.
+For those, you can additionally specify a **positive class** within the `task` object during task creation or afterwards, which is used afterwards.
+If you are not setting this manually, `mlr3` sets the positive class automatically to the first class level which appears in the target variable in the dataset.
 
-### Thresholds
+```{r 01-mlr3-basics-057}
+# directly
+data("Sonar", package = "mlbench")
+b = as_data_backend(Sonar)
+task = TaskClassif$new(id = "Sonar", backend = b, target = "Class", positive = "R")
+
+# or afterwards
+task$positive = "M"
+```
+
+### ROC Curve and Thresholds
+
+ROC Analysis -- which stands for "receiver operating characteristics -- is a subfield of Machine Learning which studies the evaluation of binary prediction systems.
+We saw earlier, that we can simply access the confusion matrix of a prediction by `prediction$confusion`, where, the positive class corresponds to the one set within the task.
+The positive class is the more important, often smaller one
+It is a representation of all predictions, where we count correct and incorrect class assignments.
+False Positive means: We assigned "positive", but were wrong.
+We can now normalize in rows and cols of the confusion matrix to derive several informative metrics in imbalanced or cost-sensitive situations:
+
+- **True Positive Rate (TPR)**: How many of the true positives did we predict as positive?
+- **True Negative Rate TNR**: How many of the true negatives did we predict as negative?
+- **Positive Predictive Value PPV**: If we predict positive how likely is it a true positive?
+- **Negative Predictive Value NPV**: If we predict negative how likely is it a true negative?
+
+```{r 01-mlr3-basics-058, echo = FALSE}
+knitr::include_graphics("images/confusion_matrix(wikipedia).png")
+```
+Source: [Wikipedia](https://en.wikipedia.org/wiki/Evaluation_of_binary_classifiers)
+
+It is difficult to achieve a high TPR and low FPR simultaneously, so we use them for constructing the ROC Curve.
```

Comment:
This logic does not really make sense? Or at least the wording is not precise enough.